robot/client/client.go

```diff
@@ -630,12 +631,18 @@ func (rc *RobotClient) resources(ctx context.Context) ([]resource.Name, []resour
 		ctx, cancel = contextutils.ContextWithTimeoutIfNoDeadline(ctx, defaultResourcesTimeout)
 		defer cancel()
 	}
+
 	resp, err := rc.client.ResourceNames(ctx, &pb.ResourceNamesRequest{})
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var resTypes []resource.RPCAPI
+
+	if rc.rpcSubtypesUnimplemented {
```

Comment:
Comment above this line would be helpful. I realize you have one ~30 lines down but nice to be clear about what we're doing here.